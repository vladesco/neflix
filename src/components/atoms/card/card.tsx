import React from 'react'
import cn from 'classnames'

import { AtomReactComponent } from '../types'
import { cardStyles } from './card.style'

export const Card: AtomReactComponent = ({ children, className }) => {
    const classes = cardStyles()

    return <div className={cn(classes.card, className)}>{children}</div>
}
