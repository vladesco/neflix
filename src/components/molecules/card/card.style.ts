import { createUseStyles } from 'react-jss'

export const cardStyles = createUseStyles({
    card: {
        display: 'inline-block',
        backgroundColor: '#232323',
        padding: '4px 16px',
        borderRadius: '10px',
        boxShadow: '0 4px 8px 0 rgba(0,0,0,0.)',
    },
})
