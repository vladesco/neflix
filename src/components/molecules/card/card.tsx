import React from 'react'
import { cardStyles } from './card.style'

export const Card: React.FC = ({ children }) => {
    const classes = cardStyles()

    return <div className={classes.card}>{children}</div>
}
