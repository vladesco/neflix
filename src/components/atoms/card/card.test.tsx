import React from 'react'
import { render } from '@testing-library/react'
import { Card } from './card'

describe('card component', () => {
    it('should be rendered', () => {
        const { container, asFragment } = render(<Card />)

        expect(asFragment()).toMatchSnapshot()
        expect(
            container.firstElementChild.className.includes('card')
        ).toBeTruthy()
    })
})
