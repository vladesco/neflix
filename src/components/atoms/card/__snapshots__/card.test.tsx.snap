// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`card component should be rendered 1`] = `
<DocumentFragment>
  <div
    class="card-0-2-1"
  />
</DocumentFragment>
`;
